server/cmwell-grid/src/main/scala/k/grid/dmap/impl/persistent/PersistentDMap.scala

```diff
@@ -73,7 +74,11 @@ class PersistentSlave extends DMapSlave with LazyLogging {
   private val dataFile = new File(s"${Grid.persistentDmapDir}/${Config.clusterName}")
 
   def readMap : Option[MapData] = {
-    val content = Try(scala.io.Source.fromFile(dataFile).getLines().mkString("\n").parseJson.convertTo[MapData]) match {
+    val content = Try {
+      val x = scala.io.Source.fromFile(dataFile).getLines().mkString("\n")
```

Comment:
This is a file leak. You must have a handle to the file and after the getLines to close it.